web_company_color/models/res_company.py

```diff
@@ -0,0 +1,165 @@
+            IrAttachmentObj.sudo().search([
+                ('url', 'like', '%s%%' % record._get_url_scss_simplified()),
+            ]).sudo().unlink()
+        return result
+
+    @api.multi
+    def write(self, values):
+        if not self.env.context.get('ignore_company_color', False):
+            fields_to_check = ('color_navbar_bg',
+                               'color_navbar_bg_hover',
+                               'color_navbar_text')
+            if 'logo' in values:
+                if values['logo']:
+                    _r, _g, _b = image_to_rgb(convert_to_image(values['logo']))
+                    # Make color 10% darker
+                    _h, _l, _s = rgb_to_hls(_r, _g, _b)
+                    _l = max(0, _l - 0.1)
+                    _rd, _gd, _bd = hls_to_rgb(_h, _l, _s)
+                    # Calc. optimal text color (b/w)
+                    # Grayscale human vision perception (rec.709 values)
+                    _a = 1 - (0.2126 * _r + 0.7152 * _g + 0.0722 * _b)
+                    values.update({
+                        'color_navbar_bg': n_rgb_to_hex(_r, _g, _b),
+                        'color_navbar_bg_hover': n_rgb_to_hex(_rd, _gd, _bd),
+                        'color_navbar_text': '#000' if _a < 0.5 else '#fff',
+                    })
+                else:
+                    values.update(self.default_get(fields_to_check))
+
+            result = super(ResCompany, self).write(values)
+
+            if any([field in values for field in fields_to_check]):
+                self.create_or_update_scss_attachment()
+        else:
+            result = super(ResCompany, self).write(values)
+        return result
+
+    @api.multi
+    def _sanitize_scss_values(self, values):
+        self.ensure_one()
+        values = dict(self.company_colors)
+        values.update({
+            'color_navbar_bg': values['color_navbar_bg'] or '$o-brand-odoo',
+            'color_navbar_bg_hover': values['color_navbar_bg_hover']
+            or '$o-navbar-inverse-link-hover-bg',
+            'color_navbar_text': values['color_navbar_text'] or '#FFF',
+        })
+        return values
+
+    @api.multi
+    def _generate_scss(self):
+        self.ensure_one()
+        # ir.attachment need files with content to work
+        if not self.company_colors:
+            return "// No Web Company Color SCSS Content\n"
+        return self.SCSS_TEMPLATE % self._sanitize_scss_values(
+            self.company_colors)
+
+    # URL to scss related with this company, without hash
+    # /web_company_color/static/src/scss/custom_colors.<company_id>
+    def _get_url_scss_simplified(self):
+        self.ensure_one()
+        NTEMPLATE = '.'.join(URL_SCSS_GEN_TEMPLATE.split('.')[:2])
+        return NTEMPLATE % self.id
+
+    @api.multi
+    def get_url_scss(self):
+        self.ensure_one()
+        return URL_SCSS_GEN_TEMPLATE % (self.id, self.scss_modif_hash)
+
+    @api.multi
+    def create_or_update_scss_attachment(self):
```

Comment:
matter of taste probably: I'd rename all the methods and the class/instance vars to be prefixed by `scss_` or `_scss`.
This way you categorize all of them automatically.
I do this when I want to "tag" and "isolate" new set of features... but feel free to ignore my comment ;)

Reply:
Done